src/Claims/DatetimeTrait.php

```diff
@@ -35,50 +36,38 @@ public function setValue($value)
     }
 
     /**
-     * Determine whether the value is numeric.
-     *
-     * @param  mixed  $value
-     *
-     * @return bool
+     * {@inheritdoc}
      */
-    protected function checkNumeric($value)
+    public function validateCreate($value)
     {
-        return is_numeric($value);
-    }
+        if (! is_numeric($value)) {
+            throw new InvalidClaimException('Invalid value provided for claim "'.$this->getName().'": '.$value);
```

Comment:
Why not a new `InvalidClaimValueException` that direct take the claim as an argument ?